RestSharp/RestClient.cs

```diff
@@ -524,6 +524,10 @@ private string EncodeParameter(Parameter parameter, Encoding encoding)
 
         internal IHttp ConfigureHttp(IRestRequest request)
         {
+            if (request.Parameters.Any(p => p.Name == null))
+            {
+                throw new ArgumentNullException("null", "Parameter name is missing. Please make sure that all the parameters names are not null.");
```

Comment:
The `ArgumentNullException` first parameter is the parameter name. Here we have dynamic parameters, I don't think it is a good idea to have this exception here. If we want to check it, it is better to throw this exception before the request is executed, in `AddXXXParameter` and there the `name` would make sense.

Reply:
I've thought about it but the problem that there are several ways to add parameter.
`request.Parameters.Add(new Parameter(...))`
`request.AddParameter(name, value)`
`request.AddParameter(new Parameter(...))`
and so on.
Adding check to every method will make it's support too complex.
I thought about adding this exception into Parameter class but it might be necessary to create a Parameter with empty name and set it later, exception will break this case.
ConfigureHttp looks like the only place where it can be solved with single condition.

Reply:
The issue is that it is much better to fail early. The logical question would be - why can I _add_ the parameter, but it fails on executing the request? And this question is completely legit.

Reply:
I'm totally agree with you but current parameters management does not allow that. Feel free to provide better PR.

Reply:
I think it does because the easiest way would be to check if the name is supplied when the new instance of the `Parameter` class is created.

Reply:
Thank you.